src/compiler/utilities.ts

```diff
@@ -4066,6 +4068,11 @@ export function isAssignmentTarget(node: Node): boolean {
     return getAssignmentTargetKind(node) !== AssignmentKind.None;
 }
 
+function isCompoundLikeAssignment(assignment: BinaryExpression): boolean {
+    const right = skipParentheses(assignment.right);
+    return right.kind === SyntaxKind.BinaryExpression && isAdditiveOperatorOrHigher((right as BinaryExpression).operatorToken.kind);

```

Comment:
perhaps this should use `isShiftOperatorOrHigher` and not `isAdditiveOperatorOrHigher`? 🤔

Reply:
Yes. For consistency with an actual compound assignment it should support a decomposed `<<=`, `>>=`, and `>>>=` as well.